app/components/createContainer/createContainerController.js

```diff
@@ -63,6 +64,15 @@ function ($scope, $state, $stateParams, $filter, Config, Info, Container, Contai
     $scope.formValues.Labels.splice(index, 1);
   };
 
+  $scope.addExtraHost = function() {
+    $scope.config.HostConfig.ExtraHosts.push({ value: '' });
```

Comment:
Instead of manipulating directly the `config.HostConfig.ExtraHosts` object, please create an `ExtraHosts` array inside the `formValues` object (this is where the temporary objects are stored) and manipulate it here.